src/score_gen/score.rs

```diff
@@ -395,6 +395,16 @@ fn scalar_to_bits(scalar: &Scalar) -> [u8; 256] {
     res
 }
 
+fn next_pow_2_bitnum(mut scalar: Scalar) -> u64 {
```

Comment:
As discussed by chat – and summarize here for future reference – the function's name "smells" because is trying to do more things than it should.
Especially if we're refactoring in #22 the implementation about how the bits are counted, we should have the two separate. Mostly, to me, this is a `bits_count` function: it just add a `+ 1` at the result. So, personally, I would just have something like:
```rust
fn bits_count(mut scalar: Scalar) -> u64 {
    scalar = scalar.reduce();
    let mut counter = 1u64;
    while scalar > Scalar::one().reduce() {
        scalar.divn(1);
        counter += 1;
    }
    counter
}
```

Notice I started `counter` from `1`, for two reasons:
- As discussed by chat, the actual condition should have been `>=` otherwise if miss the loop for `1`.
- However, if you pass `0` would have been still returned `0` – when it actually still use a bit.

Therefore keep the condition `>` but changed the initial value of `counter` fixes both scenario.

Now, we want to _round up to the next power of two_ a given Scalar. Then we have:

```rust
         let num_bits = bits_count(max_range);
         let next_pow2 = Scalar::from(2u64).pow(&[num_bits, 0, 0, 0]); 
```

Where, if you want, you can wrap these operations in a separate function than `bits_count` (e.g. `next_pow2`, that accept a `Scalar` and return a `Scalar`).